baseplate/clients/redis.py

```diff
@@ -253,3 +254,173 @@ def close(self) -> None:
         and dequeue as the actions will recreate the queue)
         """
         self.client.delete(self.queue)
+
+
+def cluster_pool_from_config(
+    app_config: config.RawConfig, prefix: str = "rediscluster.", **kwargs: Any
+) -> rediscluster.ClusterConnectionPool:
+    """Make a ClusterConnectionPool from a configuration dictionary.
+
+    The keys useful to :py:func:`clusterpool_from_config` should be prefixed, e.g.
+    ``rediscluster.url``, ``rediscluster.max_connections``, etc. The ``prefix`` argument
+    specifies the prefix used to filter keys.  Each key is mapped to a
+    corresponding keyword argument on the :py:class:`redis.ConnectionPool`
+    constructor.
+
+    Supported keys:
+
+    * ``url`` (required): a URL like ``redis://localhost/0``.
+    * ``max_connections``: an integer maximum number of connections in the pool
+    * ``timeout``: how long to wait for sockets to connect. e.g.
+        ``200 milliseconds`` (:py:func:`~baseplate.lib.config.Timespan`)
+    """
+
+    assert prefix.endswith(".")
+
+    parser = config.SpecParser(
+        {
+            "startup_nodes": config.TupleOf(config.String),
+            "max_connections": config.Optional(config.Integer, default=None),
+            "timeout": config.Optional(config.Timespan, default=None),
+        }
+    )
+
+    options = parser.parse(prefix[:-1], app_config)
+
+    nodes = [
+        {"host": node[0], "port": node[1]}
+        for node in [startup_node.split(":") for startup_node in options.startup_nodes]
+    ]
```

Comment:
It can handle the host/port parsing for us.

```suggestion
    nodes = [{"host": endpoint.address.host, "port": endpoint.address.port} for endpoint in options.startup_nodes]
```

(note `endpoint.address.port` is an integer now, is that OK for redis-py-cluster? or is it really expecting a string in that port value?)